### PR TITLE
Chore: remove array and hash tree and rename prefix.

### PR DESCRIPTION
**Pull request by jorbriib**
Array and has tree has been removed.
Prefix tree has been rename to tree.

### Refactor
Some public methods have been renamed to private methods

### CODE REVIEW

**Comment by sangarbe** [comment]
No idea about travis issue

**Comment by jorbriib** [comment]
> No idea about travis issue

Seems that github is not sending the request to travis

File: bench_router_test.go [hunk 1]
```diff
@@ -167,18 +167,8 @@ var testRoutes = []string{
...
-
-func BenchmarkMapRouter(b *testing.B) {
-	router := MapRouter{}
-	benchRouter(&router, b)
-}
-
-func BenchmarkPrefixTreeRouter(b *testing.B) {
-	router := PrefixTreeRouter{}
+func BenchmarkTreeRouter(b *testing.B) {
+	router := TreeRouter{}
```
Comment (sangarbe) [inline bench_router_test.go]: let's call it just "Router" to our router now that we don't have any other type, what to you think?

File: router.go [hunk 2]
```diff
@@ -20,64 +20,12 @@ type Router interface {
 	AddHandler(string, string, HandlerFunction)
 }
 
-type routeHandler struct {
-	path    string
-	handler HandlerFunction
+type TreeRouter struct {
```
Comment (sangarbe) [inline router.go]: As I said before, why not just "Router"?

**Review by sangarbe — DISMISSED** [review]

**Comment by jorbriib** [comment]
We already have the interface with this name (Router), what do you think if we rename treeRoute to GolossusRoute?

**Comment by sangarbe** [comment]
> We already have the interface with this name (Router), what do you think if we rename treeRoute to GolossusRoute?

Once the rest of routers are removed the interface will make no sense at all. We don't need to export that interface so the type for the router should be indeed "Router". Let's talk about this when reviewing the first release epic ;) check trello to know more.

**Review by sangarbe — APPROVED** [review]

**Pull request merged**